root/etc/e-smith/events/actions/nethserver-mattermost-conf

```diff
@@ -20,21 +20,53 @@
 # along with NethServer.  If not, see COPYING.
 #
 
+su - postgres -c "scl enable rh-postgresql12 -- psql --port=55434 -lqt | cut -d \| -f 1 | grep -q -w mattermost"
+if [[ $? -eq 0 ]]; then # mattermost database exists, exiting
+    exit 0
+fi
 su - postgres -c "scl enable rh-postgresql94 -- psql --port=55432 -lqt | cut -d \| -f 1 | grep -q -w mattermost"
-if [ $? -eq 1 ]; then # database do not exists
-
+if [[ $? -eq 0 ]]; then
+    echo "[NOTICE] Migrating data from postgresql 9.4 to postgresql 12"
+    databases=$(su - postgres -c "scl enable rh-postgresql94 -- psql -q -A -t --port=55432 -c $'SELECT datname FROM pg_database WHERE datistemplate = false AND datname <> \'postgres\';'")
+    if [[ $? -ne 0 ]]; then
+        echo "[ERROR] An error occured while retrieving list of databases"
+        exit 1
+    fi
+    systemctl stop rh-postgresql94-postgresql
+    systemctl stop rh-postgresql12-postgresql
+    su - postgres -c "export LD_LIBRARY_PATH=/opt/rh/rh-postgresql94/root/usr/lib64:/opt/rh/rh-postgresql12/root/usr/lib64:$LD_LIBRARY_PATH && scl enable rh-postgresql12 -- pg_upgrade -b /opt/rh/rh-postgresql94/root/usr/bin -B /opt/rh/rh-postgresql12/root/usr/bin -d /var/opt/rh/rh-postgresql94/lib/pgsql/data -D /var/opt/rh/rh-postgresql12/lib/pgsql/data"
+    if [[ $? -ne 0 ]]; then
+        echo "[ERROR] An error occured during postgresql migration"
+        exit 1
+    fi
+    echo "[NOTICE] Migration completed successfully"
+    if [[ "$databases" == "mattermost" ]]; then
+        # only mattermost database is present, delete all postgres 9.4 data
+        /sbin/e-smith/config setprop rh-postgresql94-postgresql status disabled
+        systemctl disable rh-postgresql94-postgresql
+        rm -rf '/var/opt/rh/rh-postgresql94/lib/pgsql/data'
+        echo "[NOTICE] postgresql 9.4 is no longer needed and has been disabled"
+    else
+        echo "[NOTICE] postgresql 9.4 seems to be still used. Removing only mattermost database"
+        systemctl start rh-postgresql94-postgresql
+        sleep 1
```

Comment:
Is there a better approach? :thinking: @gsanchietti @DavidePrincipi

Reply:
Moreover, other existing databases in 9.4 are migrated too and are present in both version 9.4 and 12 after migration. Is this correct?

Reply:
> Moreover, other existing databases in 9.4 are migrated too and are present in both version 9.4 and 12 after migration. Is this correct?

This should be really a rare corner case, I'd ignore it